File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/task_detail/TaskDetailView.java
package com.teo.ttasks.ui.activities.task_detail;

import com.teo.ttasks.data.model.TTask;
import com.teo.ttasks.data.model.TTaskList;
import com.teo.ttasks.ui.base.MvpView;

interface TaskDetailView extends MvpView {

    void onTaskLoaded(TTask task);

    void onTaskLoadError();

    void onTaskListLoaded(TTaskList taskList);

    void onTaskListLoadError();

    void onTaskUpdated(TTask task);

    void onTaskDeleted();
}

File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/task_detail/TaskDetailView.kt
package com.teo.ttasks.ui.activities.task_detail

import com.teo.ttasks.data.model.TTaskList
import com.teo.ttasks.data.model.Task
import com.teo.ttasks.ui.base.MvpView

internal interface TaskDetailView : MvpView {

    fun onTaskLoaded(task: Task)

    fun onTaskLoadError()

    fun onTaskListLoaded(taskList: TTaskList)

    fun onTaskListLoadError()

    fun onTaskUpdated(task: Task)

    fun onTaskDeleted()
}
